### PR TITLE
feat(start_planner): disable safety check against dynamic objects when ego polygon overlap with centerline

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

The issue was 
ego vehicle stops on the middle of the road when the vehicle comes from the behind. 
safety check against dynamic objects after departure was enabled.

In this PR,
disabled safety check against dynamic objects when ego polygon overlap with centerline

Before change
[Screencast from 2024年01月30日 20時17分28秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/32741405/1daf5faf-5073-45a1-8e22-865161a81906)
Ego stops against behind vehicle...

After change
[Screencast from 2024年01月30日 19時51分02秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/32741405/f834b9ef-9f77-43a4-8e9d-011cbf6ebe47)
Ego doesn't stop against behind vehicle

## Tests performed

- [x] [PASS TIERIV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/7932fe83-7c5c-5af4-86a2-941de2ea7c62?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 1]
```diff
@@ -278,6 +280,37 @@ bool StartPlannerModule::isCurrentPoseOnMiddleOfTheRoad() const
...
+      geometry_msgs::msg::Point center_point = lanelet::utils::conversion::toGeomMsgPt(point);
+      centerline_points.push_back(center_point);
+    }
+
+    for (size_t i = 0; i < centerline_points.size() - 1; ++i) {
+      const auto & p1 = centerline_points.at(i);
+      const auto & p2 = centerline_points.at(i + 1);
+      for (size_t j = 0; j < vehicle_footprint.size() - 1; ++j) {
+        const auto & p3 = tier4_autoware_utils::toMsg(vehicle_footprint[j].to_3d());
+        const auto & p4 = tier4_autoware_utils::toMsg(vehicle_footprint[j + 1].to_3d());
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:302]:
```suggestion
        const auto p3 = tier4_autoware_utils::toMsg(vehicle_footprint[j].to_3d());
        const auto p4 = tier4_autoware_utils::toMsg(vehicle_footprint[j + 1].to_3d());
```

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 2]
```diff
@@ -278,6 +280,37 @@ bool StartPlannerModule::isCurrentPoseOnMiddleOfTheRoad() const
   return std::abs(lateral_distance_to_center_lane) < parameters_->th_distance_to_middle_of_the_road;
 }
 
+bool StartPlannerModule::isOverlapWithCenterLane() const
+{
+  const Pose & current_pose = planner_data_->self_odometry->pose.pose;
+  const auto current_lanes = utils::getCurrentLanes(planner_data_);
+  const auto & local_vehicle_footprint = vehicle_info_.createFootprint();
+  const auto & vehicle_footprint =
+    transformVector(local_vehicle_footprint, tier4_autoware_utils::pose2transform(current_pose));
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:289]:
```suggestion
  const auto local_vehicle_footprint = vehicle_info_.createFootprint();
  const auto vehicle_footprint =
    transformVector(local_vehicle_footprint, tier4_autoware_utils::pose2transform(current_pose));
```

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 3]
```diff
@@ -278,6 +280,37 @@ bool StartPlannerModule::isCurrentPoseOnMiddleOfTheRoad() const
...
+      centerline_points.push_back(center_point);
+    }
+
+    for (size_t i = 0; i < centerline_points.size() - 1; ++i) {
+      const auto & p1 = centerline_points.at(i);
+      const auto & p2 = centerline_points.at(i + 1);
+      for (size_t j = 0; j < vehicle_footprint.size() - 1; ++j) {
+        const auto & p3 = tier4_autoware_utils::toMsg(vehicle_footprint[j].to_3d());
+        const auto & p4 = tier4_autoware_utils::toMsg(vehicle_footprint[j + 1].to_3d());
+        const auto intersection = tier4_autoware_utils::intersect(p1, p2, p3, p4);
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:303]: how abount just check intersection of polygon and lines not giving 4 points

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:303]: maybe it is possible with boost geometry

Comment (kosuke55) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:303]: @kyoichi-sugahara said tier4 utils is fater

Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/src/start_planner_module.cpp:303]: @kosuke55 
I'm using it because this computational cost is low

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**